src/vivarium_gates_nutrition_optimization/components/disease.py

```diff
@@ -53,24 +68,104 @@ def compute_transition_proportion(self, index):
         transition_proportion.loc[sub_pop] = self.proportion(sub_pop)
         return transition_proportion
 
-    def _probability(self, index):
+    ####################
+    # Helper methods   #
+    ####################
+
+    def _probability(self, index) -> pd.Series:
         return self.proportion_pipeline(index)
 
 
 class ParturitionExclusionState(DiseaseState):
-    def setup(self, builder: Builder):
-        super().setup(builder)
-        view_columns = self.columns_created + [self._model, "alive", "pregnancy", "tracked"]
-        self.population_view = builder.population.get_view(view_columns)
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def columns_required(self) -> List[str]:
+        return super().columns_required + ["pregnancy", "tracked"]
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def setup(self, builder: Builder) -> None:
+        """Performs this component's simulation setup.
+
+        Parameters
+        ----------
+        builder : `engine.Builder`
+            Interface to several simulation tools.
+        """
+        super(DiseaseState, self).setup(builder)
```

Comment:
What happened here that meant you had to copy/paste from `DiseaseState.setup` rather than being able just to call the direct `super().setup`?

Reply:
Right! DiseaseState in VPH from 2.0 creates the disability_weight pipeline directly in setup() instead of relegating it to a different setup method like I did in my prior refactor:
`        self.disability_weight = builder.value.register_value_producer(
            f"{self.state_id}.disability_weight",
            source=self.compute_disability_weight,
            requires_columns=["age", "sex", "alive", self.model],
        )`
but this disability weight source requires an additional column, `"pregnancy"`, so to pass it into `requires_columns` without double-registering, I thought I'd need to c/p